File: mefApi/Models/Operation.cs
using System.ComponentModel.DataAnnotations;

namespace mefApi.Models
{
    public class Operation : BaseEntity
    {
        [Required]
        public int CompteComptableId { get; set; }
        public CompteComptable? CompteComptable { get; set; }
        [Required]
        public TypeOperation TypeOperation { get; set;} = 0;
        [Required]
        public decimal Taux { get; set; } = 0;
    }
}

File: mef/src/app/model/operation.ts
import { TypeOperation } from './typeoperation';

export class Operation {
  id: number = 0;
  compteComptableId: number = -1;
  compte: string = '';
  gabaritId?: number;
  typeOperation: TypeOperation = -1;
  tOperation: string = '';
  taux?: number;
}
